src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,30 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  todo: Todo;
+  setSelectedTodoId: React.Dispatch<React.SetStateAction<number>>;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, setSelectedTodoId }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(todo.userId)
+      .then(setUser)
+      .then(() => setIsLoading(false));
+  }, [todo.userId]);
```

Comment:
Initializing state with props can lead to bugs and it is generally not recommended. The state `user` is initialized with `getUser(todo.userId)`, which is a prop. This can lead to inconsistencies in the state and the prop. Please refer to these articles for more information: [here](https://stackoverflow.com/a/50403930), [here](https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state) and [here](https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state).